### PR TITLE
Snapshot validator entries and block signers

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
* **#39** <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/39?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `feature/proof-of-stake`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lazynina and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: entries/block.go [hunk 1]
```diff
@@ -34,12 +35,43 @@ type PGBlockEntry struct {
...
+}
+
 // Convert the UserAssociation DeSo encoder to the PG struct used by bun.
-func BlockEncoderToPGStruct(block *lib.MsgDeSoBlock, keyBytes []byte, params *lib.DeSoParams) *PGBlockEntry {
+func BlockEncoderToPGStruct(block *lib.MsgDeSoBlock, keyBytes []byte, params *lib.DeSoParams) (*PGBlockEntry, []*PGBlockSigner) {
 	blockHash, _ := block.Hash()
+	blockHashHex := hex.EncodeToString(blockHash[:])
+	qc := block.Header.GetQC()
+	blockSigners := []*PGBlockSigner{}
+	if !isInterfaceNil(qc) {
```
Comment (tholonious) [inline entries/block.go:54]: Is this safe? Please make this this works for a PoW block where the header has neither a QC nor an aggregate QC

**Review by tholonious — APPROVED** [review]

File: entries/block.go (hunk 1 above)
Comment (lazynina) [reply entries/block.go:54]: yeah this is safe. the `isInterfaceNil` check returns true for pow blocks.

**Comment by lazynina** [comment]
### Merge activity

* **Apr 29, 1:06 PM EDT**: @lazynina started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/39).

**Pull request merged**